plugins/registry.js

```diff
@@ -248,6 +248,35 @@ export default class PluginRegistry {
         return id;
     }
 
+    // Register a post sub menu list item by providing some text and an action function.
+    // Accepts the following:
+    // - text - A string or React element to display in the menu
+    // - action - A function to trigger when component is clicked on
+    // - filter - A function whether to apply the plugin into the post' dropdown menu
+    // Returns an unique identifier for the root submenu, and a function to register submenu items.
```

Comment:
So I pulled down the branch and tried to work through the example, i wrestled with the api a bit to register the submenus, should we add an example here on how to register submenus, because as a plugin author it's not very obvious, actually i don't think there is a way for them to know unless they look at the code here, which isn't a good experience. Might I suggest to add an example here to help them navigate through the API?

```
const {id, rootRegisterMenuItem} = registry.registerPostDropdownSubMenuAction("root", action, filter);

rootRegisterMenuItem("sub-1", action, filter);
rootRegisterMenuItem("sub-2", action, filter);
```

Reply:
@scottleedavis

Reply:
Would the demo plugin serve as a sufficient example @ali-farooq0 ?
https://github.com/mattermost/mattermost-plugin-demo/pull/62

Reply:
personally, not a fan of that, but maybe for now that will suffice :)